portfolio/src/main/java/com/google/sps/servlets/LogServlet.java

```diff
@@ -0,0 +1,47 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import com.google.gson.Gson;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+
+@WebServlet("/log")
+public class LogServlet extends HttpServlet{
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
+        response.setContentType("application/json");
+        ArrayList<String> logInfo = new ArrayList<String>();
```

Comment:
Same advice I gave Helena, instead of using an array to return these values, create a structured object for this return data. An array requires the client to know the order the server added the values. A structured object eliminates this need and makes it extensible.